File: python/src/Behavioral/State/DvdStateContext.py
# //DvdStateContext.java - the Context

from DvdStateName import DvdStateName
from DvdStateNameStars import DvdStateNameStars

class DvdStateContext:

	dvdStateName = None

	def __init__(self): 
		self.dvdStateName = DvdStateNameStars()  # start with stars

	def setDvdStateName (self,dvdStateNameIn):
		self.dvdStateName = dvdStateNameIn

	def showName (self,nameIn):
		return self.dvdStateName.showName(self,nameIn)

File: go/behavioral/state/DvdStateContext.go

//// Original copy of this content taken from http://www.fluffycat.com/Java-Design-Patterns/ in 2010
//// Original Author: Larry Truett
//// Privacy Policy at http://www.fluffycat.com/Privacy-Policy/
//An object appears to change its' class when the class it passes calls through to switches itself for a related class.
//DvdStateContext.java - the Context

package state


type DvdStateContext struct {  
   Name DvdStateName
}


    
func NewDvdStateContext() *DvdStateContext {
	obj := new(DvdStateContext)
	obj.Name = new(DvdStateNameStars)
	return obj
}

func (this *DvdStateContext) ShowName(nameIn string) string {
	 return this.Name.ShowName(this, nameIn)
}
